community-tutorials/high-availability-keepalived/01-en.md

```diff
@@ -0,0 +1,242 @@
+language: en
+available_languages: en
+---
+
+# Introduction
+High-availability (HA) refers to the ability of a system to continue operating with a high probability despite the failure of one of its components. There are several applications that help us to achieve high-availability. In this tutorial we will use [keepalived](https://github.com/acassen/keepalived) an application for loadbalancing and high-availability, which uses the Virtual Router Redundancy Protocol (VRRP; [RFC5798](https://datatracker.ietf.org/doc/html/rfc5798)). 
+
+Keepalived is used as a daemon that monitors the system state or certain conditions to automatically failover to a standby system if an error occurs. We will additionally use a [Netcup Failover-IPv4](https://www.netcup.de/bestellen/produkt.php?produkt=1073) and a [Cloud vLAN Free](https://www.netcup.de/bestellen/produkt.php?produkt=2284).
+
+Please note that this is not an absolute beginners tutorial. I use references to other tutorials or the official documentation of used applications. If there are already instructions, I see no reason to write them again. So the main point here is how to implement a high availability setup with the mentioned products in the Netcup environment.
+
+# Requirements
+- Two Servers either [vServers](https://www.netcup.de/vserver/vps.php) or [Root Servers](https://www.netcup.de/vserver/) with Ubuntu/Debian
+- [Failover IPv4](https://www.netcup.de/bestellen/produkt.php?produkt=1073)
+- [Cloud vLAN Free](https://www.netcup.de/bestellen/produkt.php?produkt=2284) (or higher)
+- Access to your [Customer Control Panel (CCP)](https://www.customercontrolpanel.de) & [Server Control Panel (SCP)](https://www.servercontrolpanel.de)
+- Access to a user with root privileges 
+
+I assume that you have already booked the mentioned products and fulfill the stated requirements. I will simply refer to the two servers as server 1 (s1) and server 2 (s2). If I am not stating a specific server it means that the commands need to be executed on both servers.
+
+# Step 1 - Setup Cloud vLAN
+## Step 1.1 - Preparation SCP
+### Step 1.1.1 - Log in to your [SCP](https://www.servercontrolpanel.de), either directly or via the autologin feature in your [CCP](https://www.customercontrolpanel.de)
+### Step 1.1.2 - Choose a server (from the list on the top left) 
+### Step 1.1.3 - Navigate to the *Network* section (in the left sidebar)
+### Step 1.1.4 - Choose *add Ethernet Interface* (on the top right)
+### Step 1.1.5 - Select your Cloud vLAN and choose *virtio* as the driver
+### Step 1.1.6 - Select the second server and repeat the steps 1.1.3 - 1.1.5; then continue with 1.2
+## Step 1.2 - Create interface
+Log in to your servers and create a vLAN-configuration for the network interface via your favourite text editor e.g. `sudo vim /etc/network/interfaces.d/vlan.cfg`
```

Comment:
I needed to switch back from netplan to network package in order to follow your tutorial. Netplan is preinstalled in Ubuntu. Did you have to do the same?

Reply:
Yes that's true. I will check that and add a corresponding netplan config or remove Ubuntu from the supported distros.

Reply:
I removed Ubuntu from the supported distros and added a hint that the tutorial can be used as a starting point for other distributions besides Debian but may need modifications for that (https://github.com/netcup-community/community-tutorials/pull/49/commits/231e0971dfc837063dfb722a91bc1bf41bccc14a).